src/main/java/com/algorand/algosdk/v2/client/common/Client.java

```diff
@@ -0,0 +1,275 @@
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Request.Builder;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import com.algorand.algosdk.crypto.Address;
+
+import com.algorand.algosdk.v2.client.algod.AccountInformation;
+import com.algorand.algosdk.v2.client.algod.GetPendingTransactionsByAddress;
+import com.algorand.algosdk.v2.client.algod.GetBlock;
+import com.algorand.algosdk.v2.client.algod.GetSupply;
+import com.algorand.algosdk.v2.client.algod.RegisterParticipationKeys;
+import com.algorand.algosdk.v2.client.algod.ShutdownNode;
+import com.algorand.algosdk.v2.client.algod.GetStatus;
+import com.algorand.algosdk.v2.client.algod.WaitForBlock;
+import com.algorand.algosdk.v2.client.algod.RawTransaction;
+import com.algorand.algosdk.v2.client.algod.TransactionParams;
+import com.algorand.algosdk.v2.client.algod.GetPendingTransactions;
+import com.algorand.algosdk.v2.client.algod.PendingTransactionInformation;
+import com.algorand.algosdk.v2.client.indexer.SearchForAccounts;
+import com.algorand.algosdk.v2.client.indexer.LookupAccountByID;
+import com.algorand.algosdk.v2.client.indexer.LookupAccountTransactions;
+import com.algorand.algosdk.v2.client.indexer.SearchForAssets;
+import com.algorand.algosdk.v2.client.indexer.LookupAssetByID;
+import com.algorand.algosdk.v2.client.indexer.LookupAssetBalances;
+import com.algorand.algosdk.v2.client.indexer.LookupAssetTransactions;
+import com.algorand.algosdk.v2.client.indexer.LookupBlock;
+import com.algorand.algosdk.v2.client.indexer.SearchForTransactions;
+
+public class Client {
+
+	private OkHttpClient client;
+	private String host;
+	private int port;
+	private String token;
+
+	public int getPort() {
+		return port;
+	}
+
+	public String getHost() {
+		return host;
+	}
+
+	public static HttpUrl getHttpUrl(QueryData qData, int port, String host) {
+		HttpUrl.Builder httpUrlBuilder = (new HttpUrl.Builder()).scheme("http").port(port).host(host);
+		for (String ps : qData.pathSegments) {
+			httpUrlBuilder.addPathSegment(ps);
+		}
+		for (Entry<String, String> kvp : qData.queries.entrySet()) {
+			httpUrlBuilder.addQueryParameter(kvp.getKey(), kvp.getValue());
+		}
+		HttpUrl httpUrl = httpUrlBuilder.build();
+		return httpUrl;
+	}
+
+	public Client(String host, int port, String token) {
+		MediaType.parse("application/json; charset=utf-8");
+		this.host = host;
+		this.port = port;
+		this.token = token;
+		this.client = new OkHttpClient();
+	}
+
+	public Response executeCall(QueryData qData, String getOrPost) throws Exception {
+
+		HttpUrl httpUrl = getHttpUrl(qData, port, host);
+		Builder reqBuilder = new Request.Builder().url(httpUrl);
+		if (token != null) {
+			reqBuilder.addHeader("X-Algo-API-Token", token);
+		}
+		RequestBody rb = RequestBody.create(
+				MediaType.parse("Binary data"),
+				qData.bodySegments.isEmpty() ? new byte[0] : qData.bodySegments.get(0));
+
+		if (getOrPost.equals("get")) {
```

Comment:
Tsachi is adding a put/delete endpoint with catchpoints, we should probably just support all of them with the client. Could you use one of the public verb enums instead of a string? Maybe this one: https://docs.oracle.com/javaee/7/api/javax/ws/rs/HttpMethod.html

Reply:
I added a class for httpMethod types.